File: apps/backend/Ecommerce/Ecommerce.Domain/Entities/PromoCode.cs
namespace Ecommerce.Domain.Entities
{
    public class PromoCode : BaseEntity
    {
        public required string Code { get; set; }
        public required string Description { get; set; }
        public PromoCodeType Type { get; set; }
        public decimal DiscountPercentage { get; set; } // For percentage discounts
        public decimal DiscountAmount { get; set; } // For fixed amount discounts
        public bool FreeShipping { get; set; }
        public DateTime ValidFrom { get; set; }
        public DateTime ValidTo { get; set; }
        public int UsageLimit { get; set; }
        public int TimesUsed { get; set; }
        public bool IsActive { get; set; }
    }

    public enum PromoCodeType
    {
        PercentageDiscount,
        FixedAmountDiscount,
        FreeShipping,
        Mixed
    }
}

File: apps/frontend/ecommerce-dashboard/types/promo-code.ts
export interface PromoCode {
    id: string;
    code: string;
    description?: string;
    type: EPromoCodeType;
    discountPercentage: number;
    discountAmount: number;
    freeShipping: boolean;
    validFrom: Date;
    validTo: Date;
    usageLimit: number;
    timesUsed: number;
    isActive: boolean;
    isExpired: boolean;
    isAvailable: boolean;
}

export enum EPromoCodeType {
    PercentageDiscount,
    FixedAmountDiscount,
    FreeShipping,
    Mixed
}



